use-cases.md

```diff
@@ -53,11 +53,7 @@ When a server wants to trace execution of a request, it generally needs to go th
 Let's assume that we have an HTTP server, and the Span is propagated from the client via HTTP headers, accessible via `request.headers`:
 
 ```python
-    span = tracer.join_trace_from_text(
-        operation_name=operation,
-        context_snapshot=request.headers,
-        trace_attributes=request.headers
-    )
+    span = tracer.get_span_extractor(request.class).extract(request)
```

Comment:
Another issue is that it makes the tracer very inflexible. If tomorrow you decide to create your own MicksSuperFastRpcOverHttp2 framework, you won’t be able to use any of the existing OT tracers to instrument that framework, because none of them will know about MicksSuperFastHttpRequest API. Where as in the current API, any framework can use OT to instrument itself.

What I do like about this direction is an implication that the data can be written into the request potentially more efficiently than by allocating and returning two maps, then iterating through then and writing them into yet another map (headers). @bensigelman I think we should consider building better streaming support into the codec API. For example, in Go version, instead of allocating and returning two byte arrays, we could’ve taken a Writer as an argument and just write the bytes to it. That’s much more in line with std Go APIs. If people, for some reason, must get a map or blobs back, we can always provide simple wrappers to do so. For example we could define the codec as taking this as argument

```go
type BinarySpanWriter interface {
    WriteSpanID(bytes []byte)
    WriteTraceAttributes(bytes byte[])
}

type KVMapSpanWriter interface {
    WriteSpanIDKV(key, value string)
    WriteTraceAttribute(key, value string)
}
```

On the reading side, it would be done with an interator-like interface. I can create concrete examples in a separate PR.